### DISCUSSION TITLE
Fuzzy queries

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#377] [Feedback & Feature Proposal]
Because of the GitHub bug regarding discussion transfer, I create this discussion so that the product team can take into account the suggestions.

https://github.com/meilisearch/MeiliSearch/discussions/1096

**Comment by gmourier** [comment]
@eikooc, sorry to ping you here but we have transferred the discussion here!

It may be possible to configure the fuzziness behavior to your need if we expose a way to configure the typo behavior since it seems to me that's two words (`fuzziness` and `typo`) for the same thing.

E.g `Configure the minimum word size for 1 typo` / `Configure the minimum word size for 2 typos`

I imagine that it will have an impact on the performance depending on the configuration. Meilisearch is limited to 2 typos now to favor speed and relevancy for most of the use-case. May not need to have more typos allowed if you can set the trigger to a configurable minimum size of words 🤔

**Reply by eikooc** [reply]
It's true that you can get _some_ of the behavior by increasing the typo tolerance, but not exactly.

Fuzziness would match only one in this example:

```
// Documents: [{title: batman},{title: ant}]
// Query: "btm"
```

But increased typo tolerance would match them both.

**Comment by 6543** [comment]
my usecase would be to specificly set fuzziness/typo to 0 so it only return exact (case insensitive) matches

(of course fuzzy search should still be possible on that index, so per query setting)

PS: my workaround is to post-process the result and iterate over all hits and manually filter again :/

**Reply by 6543** [reply]
(https://github.com/go-gitea/gitea/pull/29671)

**Reply by macraig** [reply]
Hi @6543 , I've opened a new discussion based on your needs [here](https://github.com/orgs/meilisearch/discussions/727)
